specreduce/tracing.py

```diff
@@ -110,3 +113,164 @@ def __post_init__(self):
                 padding = np.ma.MaskedArray(np.ones(nx - nt) * self.trace[-1], mask=True)
                 self.trace = np.ma.hstack([self.trace, padding])
         self._bound_trace()
+
+
+@dataclass
+class KosmosTrace(Trace):
+    """
+    Trace the spectrum aperture in an image.
+
+    Chops image up in bins along the dispersion (wavelength) direction,
+    fits a Gaussian within each bin to determine the trace's spatial
+    center. Finally, draws a cubic spline through the bins to up-sample
+    trace along every pixel in the dispersion direction.
+
+    (The original version of this algorithm is sourced from James
+     Davenport's `kosmos` repository.)
```

Comment:
would it be useful to provide a link?  Are there any license implications?

Reply:
I'm agnostic on a link, but the original code is MIT-licensed, meaning we can modify and distribute the it without limitation. I believe the only thing we may be obligated to do is include a copy of `KOSMOS`' license here. We could add a note that it only applies to the specific file(s) that contain original or modified code from there.

We also did get [written permission](https://github.com/astropy/specreduce/issues/83#issuecomment-1050096916) from James earlier.